pkg/host-disk/host-disk.go

```diff
@@ -88,18 +89,30 @@ func dirBytesAvailable(path string) (uint64, error) {
 	return stat.Bavail * uint64(stat.Bsize), nil
 }
 
-func createSparseRaw(fullPath string, size int64) error {
+// Use named return err to allow checking Close() errors in defer
+func createSparseRaw(fullPath string, size int64) (err error) {
+	var f *os.File
 	offset := size - 1
-	f, err := os.Create(fullPath)
-	defer f.Close()
+	f, err = os.Create(fullPath)
+
 	if err != nil {
 		return err
 	}
+
+	defer closeFileAndCheckErr(f, &err)
+
 	_, err = f.WriteAt([]byte{0}, offset)
 	if err != nil {
 		return err
 	}
-	return nil
+	return err
+}
+
+func closeFileAndCheckErr(c io.Closer, err *error) {
```

Comment:
You used `io.Closer` interface, so lets generalize it. How about `closeIOAndCheckErr` ?

And use `ioErr` or just `e` inside.

Reply:
Done